fdbserver/Coordination.actor.cpp

```diff
@@ -545,15 +545,35 @@ struct LeaderRegisterCollection {
 	}
 };
 
+// extract the prefix descriptor from cluster id
+StringRef getClusterDescriptor(Key key) {
+	StringRef str = key.contents();
+	return str.eat(":");
+}
+
 // leaderServer multiplexes multiple leaderRegisters onto a single LeaderElectionRegInterface,
 // creating and destroying them on demand.
-ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf, OnDemandStore* pStore, UID id) {
+ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf,
+                                OnDemandStore* pStore,
+                                UID id,
+                                Reference<ClusterConnectionFile> ccf) {
 	state LeaderRegisterCollection regs(pStore);
 	state ActorCollection forwarders(false);
 
 	wait(LeaderRegisterCollection::init(&regs));
 
 	loop choose {
+		when(CheckDescriptorMutableRequest req = waitNext(interf.checkDescriptorMutable.getFuture())) {
+			Optional<LeaderInfo> forward = regs.getForward(req.key);
+			// Note the response returns the value of a knob enforced by checking only one coordinator. It is not
+			// quorum based.
+			if (forward.present()) {
+				req.reply.sendError(coordinators_changed());
```

Comment:
If I understand the logic on the other side of this request correctly, I don't think there's any reaction to this error and it bubbles all the way out to the caller. It doesn't look like this would be handled gracefully at the moment by `fdbcli`, so I think we have a couple options:

1. Implicitly apply the forwarding result and then try again.
2. Convert this error to some new `CoordinatorsResult::Type` in `changeQuorum` and then update `fdbcli` to print out a useful error message in response.

I think ideally we would do `#1`, as for the most part we don't expose details about forwarding to users, and I'm not sure what the remedy for the user is in this case. If that's not feasible, though, then we should at least do `#2`.

Reply:
how about we return the new  leader (forwarded info) back to the client, and it tries again. And if that fails a second time generate a useful error..

Reply:
in an offline chat with evan, it was pointed out the result of the message is the same no matter who receives it, so there is no need to forward it to the leader. i.e. this error case can be removed.